polylogue/cli/sync.py

```
@@ -33,6 +35,69 @@ def _truthy(val: str) -> bool:
     return str(val).strip().lower() in {"1", "true", "yes", "on"}
 
 
+def _apply_resume_from(args: SimpleNamespace, env: CommandEnv, *, run_id: int) -> None:
+    run = get_run_by_id(int(run_id))
+    if not run:
+        raise SystemExit(f"Unknown run id: {run_id}")
+    cmd = str(run.get("cmd") or "")
+    provider = getattr(args, "provider", None)
+    if cmd and not cmd.startswith("sync"):
+        raise SystemExit(f"Run #{run_id} is not a sync run (cmd={cmd!r}).")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)